### DISCUSSION TITLE
How to get FIDI to simply reload .env?

### DISCUSSION THREAD

**Discussion by Terry-JF** [firefly-iii/firefly-iii#9130] [Firefly III] [answered]
I have hit an issue with auto-import failing in a multi-user environment. I created an xxx_import.json for the bank related to a 2nd user and put this into the /import directory, but of course it does not work, things are never that simple. The FIDI .env has the PAT of user 1 so though FIDI does get the transactions from GoCardless, FIDI is connected to Firefly as user 1 and so cannot 'see' the accounts belonging to user 2, the user 1 bank transactions continue to load as normal.

So my thinking is... I set up a '.env1' with user 1 details and a '.env2' with user 2 details. I also create a /import2 directory and put the .json files for user 2 in there. Then in the www-data crontab I arrange to...

- at say 7am
- copy .env1 to .env
- get FIDI to reload the .env file
- run the auto-import from /import
- at say 7:30am
- copy .env2 to .env
- get FIDI to reload the .env file
- run the auto-import from /import2

Does this make sense as a simple solution, and if so what is the easiest way to get FIDI to reload the .env file? (I'm sure I could eventually figure out how to stop and restart FIDI from the cron script, but there is probably something simpler).

Thanks, Terry

**Comment by Terry-JF** [comment]
I forgot that FF3 and FIDI are web apps, so I think simply restarting apache2 will do what I want. I will give this a try.

**Reply by JC5** [reply]
You could also consider using Docker, you can even run two instances at the same time.

**Comment by Terry-JF** [comment, marked as answer]
Just to add that the simple solution is often the best, so for the benefit of anybody in a similar situation in the future...
I edited the crontab for the www-data user as outlined in my original post and this works fine, as shown below:

```shell
0 6 * * * cp /var/www/html/data-importer/.env2 /var/www/html/data-importer/.env
5 6 * * * /bin/php /var/www/html/data-importer/artisan importer:auto-import /var/www/html/data-importer/import2
20 6 * * * cp /var/www/html/data-importer/.env1 /var/www/html/data-importer/.env
25 6 * * * /bin/php /var/www/html/data-importer/artisan importer:auto-import /var/www/html/data-importer/import
``` 

The .json files for the original (first) FF3 user's banks are in /import and the .json files for the 2nd FF3 user's accounts are in /import2. 
The .env1 and .env2 files are simply copies of the original .env but with the Personal Access Token for user2 inserted to replace that for user1 in the .env2 file.

(Edit) It is not necessary to restart apache2 as the .env file is read by FIDI each time you run the auto-import routine.

I hope this helps somebody sometime.
